oqs-sys/build.rs

```diff
@@ -117,7 +114,25 @@ fn main() {
     }
     println!("cargo:rustc-link-search=native={}", libdir.display());
 
-    let gen_bindings = |file, filter| generate_bindings(&outdir, file, filter);
+    outdir
+}
+
+fn main() {
+    // Check if clang is available before compiling anything.
+    bindgen::clang_version();
+
+    let config = pkg_config::Config::new()
+        .range_version("0.8.0".."0.9.0")
+        .probe("liboqs");
+    let includedir = match config {
+        Ok(lib) => lib.include_paths.first().cloned().unwrap(),
+        _ => {
```

Comment:
rather than letting this depend on successfully finding the system library, I think I would rather use a `vendored` feature like `openssl-sys`.

Reply:
I'll add a `vendored` feature to both crates. My idea is, if `vendored` is enabled, the crate forces vendored sources, otherwise it checks for system library but still allows fallback to the vendor.

Note that the user can also force vendor via the `LIBOQS_NO_PKG_CONFIG` env var as specified in the pkg_config crate